### PR TITLE
Sort works in dev environment and persists even if a task is assigned to multiple developers

### PR DESCRIPTION
**Pull request by markediez**
1. Range slider works
2. Sort works in dev environment 
3. Assignments have their own sort position
4. Cleaned up datepicker params

### CODE REVIEW

File: app/assets/javascripts/assignments.js.coffee [hunk 1]
```diff
@@ -132,12 +134,20 @@ setupDragAndDropForAssignments = () ->
       tasks.each () ->
         taskIds.push $(this).data("task-id")
 
+      developerId = $($(this).closest("[data-developer-id]")).data("developer-id")
       # Save sort order
-      $.post
-        url: "/tasks/sort"
-        data:
-          task:
+      if developerId == -1
```
Comment (cthielen) [inline app/assets/javascripts/assignments.js.coffee]: Add success and failure callbacks for any $.ajax / $.post / $.get usage

File: app/assets/javascripts/assignments.js.coffee [hunk 2]
```diff
@@ -157,21 +167,23 @@ setupDragAndDropForAssignments = () ->
 
 setupDatePicker = () ->
   # Get time in seconds
-  time = window.location.search.substr(1).split("=")[1] * 1000
+  time = window.location.search.substr(1).split("=")[1]
```
Comment (cthielen) [inline app/assets/javascripts/assignments.js.coffee]: Can we extract this info from Ruby/.erb instead?

Comment (cthielen) [inline app/assets/javascripts/assignments.js.coffee]: 'viewDate' or 'assignmentsDate' or something else would be better than 'time', which is too generic

File: app/assets/javascripts/assignments.js.coffee [hunk 3]
```diff
@@ -157,21 +167,23 @@ setupDragAndDropForAssignments = () ->
...
     format: "DD, M d, yyyy"
     autoclose: true
 
   # Set default value
-  if isNaN time
-    time = Date.now()
+  if time
+    time = new Date time.split("-").join("/")
+  else
+    time = new Date Date.now()
```
Comment (cthielen) [inline app/assets/javascripts/assignments.js.coffee]: Doesn't new Date() default to Date.now()?

File: app/assets/javascripts/assignments.js.coffee [hunk 4]
```diff
@@ -185,3 +197,10 @@ setupDatePicker = () ->
     currDate.setDate(currDate.getDate() - 1)
     $(".date-picker").datepicker('setDate', currDate)
     $(".date-picker").datepicker('update')
+
+getDateString = (date) ->
```
Comment (cthielen) [inline app/assets/javascripts/assignments.js.coffee]: Comment this function

File: app/controllers/assignments_controller.rb [hunk 5]
```diff
@@ -2,11 +2,35 @@ class AssignmentsController < ApplicationController
...
+    @time_to_view = params[:time].present? ? DateTime.parse(params[:date]) : Time.now
 
     @developers = Developer.where(:active => true).order(created_at: :desc)
 
     authorize! :manage, @developers
     authorize! :manage, @activity
   end
+
+  # POST /assignments/sort
+  def sort
```
Comment (cthielen) [inline app/controllers/assignments_controller.rb]: Would this make more sense as an Assignments#update? If so, let's do that as it's always better to stick with the REST convention when we can than deploy custom methods.

File: app/controllers/assignments_controller.rb [hunk 6]
```diff
@@ -2,11 +2,35 @@ class AssignmentsController < ApplicationController
...
+    # Grab the developer and tasks
+    developer_account_id = params["developer_account_id"]
+    tasks = params["task_ids"]
+    positions = (0...tasks.size).to_a
+
+    # Use raw SQL to update because rails does not support cases
+    sql = "UPDATE assignments SET sort_position = CASE task_id "
```
Comment (cthielen) [inline app/controllers/assignments_controller.rb]: Why do you need CASE?

File: app/controllers/tasks_controller.rb [hunk 7]
```diff
@@ -123,11 +123,7 @@ def unassign
       developer = developer_account.developer
       task_id = params[:task][:task_id].to_i
 
-      developer.assignments.each do |a|
-        if a.task_id == task_id
-          a.delete
-        end
-      end
+      Assignment.where(:developer_account_id => developer_account.id).where(:task_id => task_id).first.delete
```
Comment (cthielen) [inline app/controllers/tasks_controller.rb]: What's this line for?

File: db/migrate/20170207210258_add_sort_position_to_assignments.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+class AddSortPositionToAssignments < ActiveRecord::Migration[5.0]
+  def change
+    add_column :assignments, :sort_position, :int, :default => 0, :null => false
```
Comment (cthielen) [inline db/migrate/20170207210258_add_sort_position_to_assignments.rb]: What's the logic behind defaulting to 0 and disallowing null? Could an assignment exist without a sort_position if it's newly imported and has no opinion on its position?

File: app/controllers/tasks_controller.rb (hunk 7 above)
Comment (markediez) [inline app/controllers/tasks_controller.rb]: It replaces the code removed. Deletes the assignment

File: db/migrate/20170207210258_add_sort_position_to_assignments.rb (hunk 8 above)
Comment (markediez) [inline db/migrate/20170207210258_add_sort_position_to_assignments.rb]: No particular logic... I guess? I just generally think sort starts at 0 and would never be null. However, NULL could just be translated as 0

File: app/controllers/assignments_controller.rb (hunk 6 above)
Comment (markediez) [inline app/controllers/assignments_controller.rb]: To assign the correct sort_position for a task_id.

File: app/controllers/assignments_controller.rb (hunk 5 above)
Comment (markediez) [inline app/controllers/assignments_controller.rb]: Sure thing

File: app/assets/javascripts/assignments.js.coffee (hunk 3 above)
Comment (markediez) [inline app/assets/javascripts/assignments.js.coffee]: Yes, yes it does

File: app/assets/javascripts/assignments.js.coffee (hunk 2 above)
Comment (markediez) [inline app/assets/javascripts/assignments.js.coffee]: It doesn't seem like I can inject rails in the js files.

When I set the following in the assignments controller
```Ruby
@viewDate = params[:date]
```

Then proceed to use it in assignments.js.coffee (or even when I rename it to assignments.js.coffee.erb)
```Ruby
viewDate = <%= @viewDate %>
console.log viewDate
```

viewDate comes out as undefined

```

File: db/migrate/20170207210258_add_sort_position_to_assignments.rb (hunk 8 above)
Comment (markediez) [inline db/migrate/20170207210258_add_sort_position_to_assignments.rb]: A null value for sort_position creates a problem when sorting. It seems like the `<=>` operator when I sort breaks when the value is nil

**Pull request merged**